### PR TITLE
Add TravisCI test-build-deploy. #5

### PR DESCRIPTION
**Pull request by rakannimer**
Hey ! 

I setup the config to get CircleCI to test the project. 
It might help in reviewing PRs as it gets more popular and more tests are added. 

Feel free to dismiss if you don't agree.

### CODE REVIEW

**Comment by kitze** [comment]
Thanks @rakannimer!

I had this open: https://github.com/kitze/JSUI/issues/5

Any ideas why you went with Circle CI? I haven't used both of them so I'm curious if you chose that one for any particular reason.

Thanks

**Comment by rakannimer** [comment]
Hey!

Sorry I missed that issue, no reason at all to use circleci in particular, I'm just used to it but have used both of them.

If you already have travis setup, let's close this PR and I'll open a new one with travis?

Cheers :)

**Comment by rakannimer** [comment]
Let me also take a look at generating and publishing builds from the CI provider. Circle/Travis. Will post back here with results.
Cheers!

**Comment by rakannimer** [comment]
I updated the CI setup to Travis. 

The code needs a mac to build, CircleCI only offers mac machines to paid plans. TravisCI offers them for free for open-source projects. 

To get this working you're going to need to add GITHUB_API_KEY as an env variable on your Travis project with a Github personal access token, to allow it to release builds for every commit on master.

Here are [some examples of generated builds.](https://github.com/rakannimer/JSUI/releases)

The title shows up as untagged-{shakey} because the commits were untagged.

If a tag is detected it will be used as the release name.

**Comment by kitze** [comment]
Thanks @rakannimer I'll take a look at this and merge it tomorrow.

**Comment by rakannimer** [comment]
Cheers ! Take your time :)

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+language: node_js
+os: osx
+node_js:
+  - "8.9.4"
```
[the remark this answers is not in the document]
Comment (rakannimer) [reply .travis.yml]: Hey @DanielRuf,

[the remark this answers is not in the document]
Comment (rakannimer) [reply .travis.yml]: Hey @DanielRuf,

Thank you for your interest in this PR,

8.9.4 is the minimum node version for the build to work.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+language: node_js
+os: osx
```
[the remark this answers is not in the document]
Comment (rakannimer) [reply .travis.yml]: In order to be able to build the electron binary for osx we need to build it on a osx machine (classic Apple)

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielRuf) [reply .travis.yml]: But this locks it to 8.9.4. Just use 8 to use the latest release of the stable branch.

File: .travis.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (DanielRuf) [reply .travis.yml]: So this just for the build / publish job? We might need a build matrix with Linux and macOS.

**Comment by rakannimer** [comment]
This PR should probably be closed in favor of @david-daly more recent one 
 : https://github.com/kitze/JSUI/pull/101

**Comment by kitze** [comment]
Tnx @rakannimer and sorry for the delay, accepted the other one 👍

**Pull request closed without merging**